PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetExtractor.cxx

```diff
@@ -1345,29 +1344,26 @@ bool AliAnalysisTaskJetExtractor::PerformGeometricalJetMatching(AliJetContainer&
   return true;
 }
 //________________________________________________________________________
-void AliAnalysisTaskJetExtractor::GetJetTagging(AliJetContainer& contBase, AliJetContainer& contTag, Bool_t isMatchTwo,  Double_t& detJetPt, Double_t& partJetPt)
+void AliAnalysisTaskJetExtractor::GetJetTagging(AliEmcalJet* jet, AliJetContainer& hybridJetCont,  AliJetContainer& detJetCont, AliJetContainer& partJetCont,  Double_t& detJetPt, Double_t& partJetPt)
 {
-  for (auto jet1 : contBase.accepted()) {
-    // Setup
-    Double_t ptJet1 = jet1->Pt() - contBase.GetRhoVal() * jet1->Area();
-    //std::cout << "pTJet1: " << ptJet1 << std::endl;
-    // Retrieve jet 2
-    AliEmcalJet * jet2 = jet1->ClosestJet();
-    if (!jet2) { continue; }
-    AliDebugStream(4) << "jet2: " << jet2->toString() << "\n";
-    Double_t ptJet2 = jet2->Pt() - contTag.GetRhoVal() * jet2->Area();
-    // This will retrieve the fraction of jet2's momentum in jet1.
-    Double_t fraction = contBase.GetFractionSharedPt(jet1);
-    //std::cout << "fraction: " << fraction << std::endl;
-    if (fraction < 0.5 && !(isMatchTwo)) {
-      continue;
-    }
-    //    std::cout << "tagged Jet pT: " << matchedJetPt << std::endl;
-    if(isMatchTwo){
-      detJetPt  = ptJet1;
-      partJetPt = ptJet2;
-    }
+  // hybrid to detector level matching 
+  AliEmcalJet * jet2 = jet->ClosestJet();
+  if (!jet2) { return; }// if there is no match return.
+  Double_t ptJet2 = jet2->Pt() - detJetCont.GetRhoVal() * jet2->Area();
+  // This will retrieve the fraction of jet2's momentum in jet1.
+  Double_t fraction = hybridJetCont.GetFractionSharedPt(jet);
+  if (fraction < 0.5) {
+    return;
   }
+  // detector to particle matching
+  AliEmcalJet * jet3 = jet2->ClosestJet();
+  if(!jet3){return;}
+  Double_t ptJet3 = jet3->Pt() - partJetCont.GetRhoVal() * jet3->Area(); 
```

Comment:
Just to point out - partJetCont should always have a Rho of 0, right? (Unless you're doing some sparse subtraction, I guess?). No need to change, but just to think about

Reply:
Yes it should!